mvlearn/merge/merge_transformers.py

```diff
@@ -0,0 +1,283 @@
+"""Merging utilities."""
+# Copyright 2019 NeuroData (http://neurodata.io)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# Authors: Pierre Ablin
+
+import numpy as np
+
+from abc import abstractmethod
+
+from sklearn.base import TransformerMixin
+from sklearn.utils.validation import check_is_fitted
+
+from ..utils.utils import check_Xs
+
+
+class BaseMerger(TransformerMixin):
+    """A base class for merging multiview datasets into single view datasets.
+
+    The .transform function should return a single dataset.
+
+    Parameters
+    ----------
+    Attributes
+    ----------
+    See Also
+    --------
+    """
+    def __init__(self):
+        pass  # pragma: no cover
+
+    @abstractmethod
+    def fit(self, Xs, y=None):
+        r"""Fit model to multiview data.
+
+        Parameters
+        ----------
+        Xs: list of array-likes
+            - Xs shape: (n_views,)
+            - Xs[i] shape: (n_samples, n_features_i)
+        y : array, shape (n_samples,), optional
+
+        Returns
+        -------
+        self: returns an instance of self.
+        """
+
+        return self  # pragma: no cover
+
+    @abstractmethod
+    def transform(self, Xs, y=None):
+        r"""Merge multiview data into a single dataset
+
+        Parameters
+        ----------
+        Xs: list of array-likes
+            - Xs shape: (n_views,)
+            - Xs[i] shape: (n_samples, n_features_i)
+        y : array, shape (n_samples,), optional
+
+        Returns
+        -------
+        X_transformed : numpy.ndarray of shape (n_samples, n_features)
+             The singleview output
+        """
+        pass  # pragma: no cover
+
+    @abstractmethod
```

Comment:
Don't think this should be abstract as it can function as is for inheriting classes.